### ISSUE TITLE
[BUG] v2 does not include bundleDependencies if has files

### ISSUE THREAD

**Issue by tsondergaard** [npm/npm-packlist#44]
I have a package.json file with a "files" section *and* "bundleDependencies". I use this to package everything that I need for "production" which excludes e.g. some tests. With v1.4.6 this works. It stopped working with v2.0.0 and still does not work with v2.0.1.

I added the following test to my clone of npm-packlist to demonstrate the problem. If I alter the test so the package.json does not have "files" then it packages bundleDependencies.

```
From 8594de0245c5a33571599649abcf42c728f29bbf Mon Sep 17 00:00:00 2001
From: Thomas Sondergaard <tsondergaard@vitalimages.com>
Date: Tue, 26 Nov 2019 20:19:23 +0100
Subject: [PATCH] Add failing test for package.json with "files" and
 "bundleDependencies"

---
 ...age-json-files-with-bundled.js-TAP.test.js | 20 +++++++++++
 test/package-json-files-with-bundled.js       | 36 +++++++++++++++++++
 2 files changed, 56 insertions(+)
 create mode 100644 tap-snapshots/test-package-json-files-with-bundled.js-TAP.test.js
 create mode 100644 test/package-json-files-with-bundled.js

diff --git a/tap-snapshots/test-package-json-files-with-bundled.js-TAP.test.js b/tap-snapshots/test-package-json-files-with-bundled.js-TAP.test.js
new file mode 100644
index 0000000..1e4e562
--- /dev/null
+++ b/tap-snapshots/test-package-json-files-with-bundled.js-TAP.test.js
@@ -0,0 +1,20 @@
+/* IMPORTANT
+ * This snapshot file is auto-generated, but designed for humans.
+ * It should be checked into source control and tracked carefully.
+ * Re-generate by setting TAP_SNAPSHOT=1 and running tests.
+ * Make sure to inspect the output below.  Do not ignore changes!
+ */
+'use strict'
+exports[`test/package-json-files-with-bundled.js TAP package with files and bundledDependencies > expect resolving Promise 1`] = `
+Array [
+  "lib/one.js",
+  "package.json",
+]
+`
+
+exports[`test/package-json-files-with-bundled.js TAP package with files and bundledDependencies > must match snapshot 1`] = `
+Array [
+  "lib/one.js",
+  "package.json",
+]
+`
diff --git a/test/package-json-files-with-bundled.js b/test/package-json-files-with-bundled.js
new file mode 100644
index 0000000..7f27843
--- /dev/null
+++ b/test/package-json-files-with-bundled.js
@@ -0,0 +1,36 @@
+const t = require('tap')
+
+const elfJS = `
+module.exports = elf =>
+  console.log("i'm a elf")
+`
+
+const pkg = t.testdir({
+  'package.json': JSON.stringify({
+    xfiles: [
+      '/lib/one.js',
+    ],
+    'bundleDependencies': [
+      'history'
+    ]
+  }),
+  lib: {
+    'one.js': 'one',
+  },
+  node_modules: {
+      history: {
+        'package.json': JSON.stringify({
+          name: '@npmwombat/history',
+            version: '1.0.0',
+          main: 'index.js'
+        }),
+        'index.js': elfJS,
+      }
+    }
+})
+
+const packlist = require('../')
+t.test('package with files and bundledDependencies', async t => {
+  t.matchSnapshot(packlist.sync({path: pkg}))
+  await t.resolveMatchSnapshot(packlist({path: pkg}))
+})
-- 
2.21.0


```

**Comment by isaacs** [comment]
Fixed by #46.

**Comment by isaacs** [comment]
Er, rather, not by #46, but by https://github.com/npm/npm-packlist/commit/d9b751e005244631e648839608415301f8f6e640, which I landed when I ran into this issue myself, without realizing that there was an open PR for it.
